packages/runtime-dom/types/jsx.d.ts

```diff
@@ -1301,7 +1301,7 @@ export interface Events {
 }
 
 type EventHandlers<E> = {
-  [K in keyof E]?: E[K] extends Function ? E[K] : (payload: E[K]) => void
+  [K in keyof E]?: E[K] extends (...args: any) => any ? E[K] : (payload: E[K]) => void
```

Comment:
This is IMO a good change anyways as there is even a eslint rule to warn using `Function` at all